changes/318.bugfix.rst

```diff
@@ -0,0 +1,6 @@
+This PR fixes JP-3791, JP-3771, and JP-3708.  For `ramp_fitting`, the
+`CRMAG` element was not originally implementd in the C-extension for
+ramp fitting.  It is now implemented.  The bug the read noise recalculation
+for CHARGELOSS when using the multiprocessing option has been fixed.  Further,
```

Comment:
Why "other than NIRISS"?  This CHARGELOSS recalculation is done in NIRISS.  I chose `test_niriss_image.py` test because the data used for the `rate` tests contain CHARGELOSS flags.

Reply:
Am I misremembering?  I thought the issue from the user's perspective was that multiprocessing + any instrument other than NIRISS threw the `pr->orig_gdq is NULL` error, because chargeloss corrections were not initialized properly.

Reply:
The error thrown during multiprocessing because `pr->orig_gdq` occurred because the CHARGELOSS flag was set, but the data got sliced the `orig_gdq` didn't get sliced and was always always passed to the C-extension as `NULL`.  Below is the fix for that.

https://github.com/spacetelescope/stcal/blob/ebf46dd2023c1b2c23ee97badc93c5928786844e/src/stcal/ramp_fitting/ols_fit.py#L581

Reply:
Okay, you can just leave that suggestion out then!

Reply:
You want me to leave out that this PR fixes a bug found in multiprocessing?